src/main/java/com/google/sps/data/friend_graph/UserFriendsGraph.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data.friend_graph;
+
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.HashMap;
+
+/*
+* A graph of all the friendships between users
+*/
+public class UserFriendsGraph {
```

Comment:
Ok, can you explain to me what is essentially the goal of the project at this point?

If the goal is to match 2nd friends (i.e., you + me are friends, but you are friends with Alisha so you want to recommend Alisha as my friend), I'm starting to fail to see why the edges are necessary.

In this class you create a hash map from every user to a set of their friends. If you want to find matches for two users (again, let's say you+me), can't you just diff the sets of friends for you and me? If your set is <Alisha, Jess> and mine is <Jess>, then set difference(your friends, my friends) should return <Alisha>. Is that not enough information?

Reply:
The goal of the project is still to connect 2nd friends. The hash map in this class was going to be used to help find the 2nd friends for each user. The list of 2nd friends that are found are not considered matches yet, but these are potential matches. Once the list of 2nd friends are found for a particular user, each of the 2nd friend's profiles will appear on the user's feed one at a time, and then the user can choose whether to friend or pass on that 2nd friend. If both the user and the 2nd friend choose to friend each other, that is considered a match. Users who have matched will then be directed to friend and communicate with each other if they choose.

My intention with the additions in this pull request was to create the hash map to map out all of the Facebook friendships (the purpose of the UserFriendsGraph.java file) which I will use later to go through and find all of the 2nd friends for each user. I intended to use a map here so that there would be easy access the friend sets for each user. I'm realizing now that I was making the process of finding these 2nd friends too complicated with creating the FriendRelationship to represent edges, because this part isn't really necessary anymore. I can remove the FriendRelationship/edges here, and when I go to write the method that will go through the map created in UserFriendsGraph.java, I can diff the sets of friends between a user and each of their friends.

I'm going to go back and re-work the files on this pull request to reflect this.